pkg/webhook/plugins/nodeaffinitywithcache/node_affinity_with_cache.go

```diff
@@ -78,10 +76,10 @@ func (p *NodeAffinityWithCache) Mutate(pod *corev1.Pod, runtimeInfos map[string]
 		return
 	}
 
-	tieredLocality, err := p.getTieredLocality()
-	if err != nil {
-		log.Error(err, "get tiered locality config error")
-		return shouldStop, err
+	tieredLocality := p.tieredLocality
+	if tieredLocality == nil {
```

Comment:
I suggest making the failure as early as possible. I think we can make it panic if tieredLocality is empty.